OsmAnd/src/net/osmand/plus/plugins/externalsensors/dialogs/SelectExternalDeviceFragment.java

```diff
@@ -259,18 +256,18 @@ public void onDeviceDisconnect(@NonNull AbstractDevice<?> device) {
 	public void onSensorData(@NonNull AbstractSensor sensor, @NonNull SensorData data) {
 	}
 
-	public void onDeviceSelected(@Nullable AbstractDevice<?> device) {
+	public void onDeviceSelected(@Nullable String deviceId) {
 		if (getTargetFragment() instanceof SelectDeviceListener) {
-			((SelectDeviceListener) getTargetFragment()).selectNewDevice(device, widgetDataFieldType);
+			((SelectDeviceListener) getTargetFragment()).selectNewDevice(deviceId, widgetDataFieldType);
 			requireActivity().onBackPressed();
 		}
 	}
 
 	enum States {
-		NO_BLUETOOTH, NOTHING_FOUND, CONTENT
+		NO_BLUETOOTH, CONTENT
```

Comment:
Can we guarantee that we always find something?

Reply:
if nothing found, we have None variant. it was one of the point in task to remove empty view